collect_app/src/main/java/org/odk/collect/android/application/Collect.java

```diff
@@ -213,10 +213,12 @@ public String getVersionedAppName() {
      * This deviates from the recommended format as described in https://github.com/opendatakit/collect/issues/3253.
      */
     public String getUserAgentString() {
```

Comment:
I think it'd be really nice to pull this method into it's own helper/use case object (`AndroidUserAgent` for instance that implements a `UserAgent` interface). That would let us pass this object to where it's needed and easily fake/mock it in cases where we might not have all the static state set up (that might be the fix for the broken tests). 

I'd considered also having it be tested but I think the test and the code would end up basically being the same formatting code.

Reply:
Hie @seadowg Thanks for the suggestion and sorry for late replying. I think the tests failed because I tried to split the host identifier. But yes we can have this method in a separate helper class `AndroidUserAgent` with a `User Agent` interface. Should I go about it?

Reply:
Yeah I think it's a nice bit of restructuring we can do as part of this work.